Index: src/elements/AmbientLight.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package elements;\r\n\r\nimport primitives.Color;\r\n\r\n/**\r\n * A class that represents ambient light of the scene\r\n */\r\npublic class AmbientLight {\r\n    Color _intensity;\r\n\r\n    /**\r\n     * the AmbientLight constructor using _intensity and _kA parameters\r\n     * @param _intensity intensity filling lighting\r\n     */\r\n    public AmbientLight(Color _intensity) {\r\n        this._intensity = _intensity;\r\n    }\r\n\r\n    /**\r\n     * a function allowing to  return the value of the Color ambient lighting intensity.\r\n     * @return the value of the Color ambient lighting intensity.\r\n     */\r\n    public java.awt.Color getIntensity(){\r\n        return _intensity.getColor();\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/AmbientLight.java	(revision f24d07eca2a4e6a7f76cb027772ec5a111ad1dfe)
+++ src/elements/AmbientLight.java	(date 1588926967180)
@@ -5,22 +5,19 @@
 /**
  * A class that represents ambient light of the scene
  */
-public class AmbientLight {
-    Color _intensity;
+public class AmbientLight extends Light {
 
     /**
      * the AmbientLight constructor using _intensity and _kA parameters
      * @param _intensity intensity filling lighting
+     * @param kA = the scale of color
      */
-    public AmbientLight(Color _intensity) {
-        this._intensity = _intensity;
+    public AmbientLight(Color _intensity, double kA) {
+        super(_intensity.scale(kA));
     }
 
     /**
      * a function allowing to  return the value of the Color ambient lighting intensity.
      * @return the value of the Color ambient lighting intensity.
      */
-    public java.awt.Color getIntensity(){
-        return _intensity.getColor();
-    }
 }
Index: src/elements/SpotLight.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/SpotLight.java	(date 1588928128509)
+++ src/elements/SpotLight.java	(date 1588928128509)
@@ -0,0 +1,7 @@
+package elements;
+
+import primitives.Vector;
+
+public class SpotLight extends PointLight {
+    private Vector _direction;
+}
Index: src/elements/PointLight.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/PointLight.java	(date 1588928128498)
+++ src/elements/PointLight.java	(date 1588928128498)
@@ -0,0 +1,8 @@
+package elements;
+
+import primitives.Point3D;
+
+public class PointLight extends Light implements LightSource {
+    protected Point3D _position;
+    protected double _kC, _kL, _kQ;
+}
Index: src/elements/DirectionalLight.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/DirectionalLight.java	(date 1588928128503)
+++ src/elements/DirectionalLight.java	(date 1588928128503)
@@ -0,0 +1,7 @@
+package elements;
+
+import primitives.Vector;
+
+public class DirectionalLight extends Light implements LightSource {
+    private Vector _direction;
+}
Index: src/elements/LightSource.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/LightSource.java	(date 1588927075289)
+++ src/elements/LightSource.java	(date 1588927075289)
@@ -0,0 +1,11 @@
+package elements;
+
+import primitives.Color;
+import primitives.Point3D;
+import primitives.Vector;
+
+public interface LightSource {
+    public Color getIntensity(Point3D p);
+    public Vector getL(Point3D p);
+
+}
Index: src/elements/Light.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/elements/Light.java	(date 1588927017667)
+++ src/elements/Light.java	(date 1588927017667)
@@ -0,0 +1,15 @@
+package elements;
+
+import primitives.Color;
+
+abstract class Light {
+    protected Color _intensity;
+
+    public Light(Color _intensity) {
+        this._intensity = _intensity;
+    }
+
+    public Color get_intensity() {
+        return _intensity;
+    }
+}
